opds_import.py

```diff
@@ -1,60 +1,22 @@
-from core.opds_import import OPDSImporter
-from core.s3 import S3Uploader
+from core.opds_import import OPDSImporterWithS3Mirror
 from core.model import (
     Representation,
-    Hyperlink,
-    DataSource,
 )
-from urllib import urlopen
 from copy import deepcopy
 from nose.tools import set_trace
 
-class ContentOPDSImporter(OPDSImporter):
-    """OPDS Importer that mirrors content to S3."""
-
-    def import_from_feed(self, feed, even_if_no_author=False):
-        imported, messages, next_links = super(ContentOPDSImporter, self).import_from_feed(feed, even_if_no_author=even_if_no_author)
-
-        # Mirror books and images to S3 for the imported editions
-        uploader = S3Uploader()
-        for edition in imported:
-            links = edition.primary_identifier.links
-            for link in links:
-                representation = link.resource.representation
-                media_type = representation.media_type
-                original_url = link.resource.url
-
-                mirror_url = None
-                if media_type in Representation.BOOK_MEDIA_TYPES:
-                    extension = Representation.FILE_EXTENSIONS.get(media_type)
-                    mirror_url = uploader.book_url(edition.primary_identifier, extension=extension, title=edition.title)
-                elif link.rel == Hyperlink.THUMBNAIL_IMAGE or link.rel == Hyperlink.IMAGE:
-                    data_source = DataSource.lookup(self._db, self.data_source_name)
-                    mirror_url = uploader.cover_image_url(data_source, edition.primary_identifier, original_url.split("/")[-1])
-               
-                if mirror_url:
-                    try:
-                        representation.mirror_url = mirror_url
-                        handle = urlopen(original_url)
-                        content = handle.read()
-                        handle.close()
-                        representation.content = content
-                        uploader.mirror_one(representation)
-                        representation.content = None
-                        edition.work.set_presentation_ready()
-                    except (IOError, TypeError) as e:
-                        print "Unable to mirror %s" % original_url
-                        representation.mirror_exception = str(e)
-                        representation.mirror_url = None
-
-        return imported, messages, next_links
-
-
-class UnglueItOPDSImporter(ContentOPDSImporter):
+class UnglueItOPDSImporter(OPDSImporterWithS3Mirror):
     """Importer for unglue.it OPDS feed, which has acquisition links from multiple sources for some entries."""
 
-    def import_from_feed(self, feed):
-        return super(UnglueItOPDSImporter, self).import_from_feed(feed, even_if_no_author=True)
+    def import_from_feed(
+            self, feed, even_if_no_author=False, 
+            cutoff_date=None, immediately_presentation_ready=False
```

Comment:
This is common code called by OPDSImportMonitor.follow_one_link, so I couldn't change the signature. I've changed the defaults to reflect the values that this subclass uses.

I think that even_if_no_author, cutoff_date, and immediately_presentation_ready have become kind of miscellaneous and should go into a new Policy object (or even into ReplacementPolicy, I'm not sure). I think it's the same divide I mentioned earlier--either we are importing the actual books (in which case they should be made presentation ready ASAP even if there is no known author) or we are importing metadata (in which case we should hold off until we get all the data).

Does this answer your question?